internal/app/machined/pkg/runtime/v1alpha1/bootloader/grub/probe.go

```diff
@@ -0,0 +1,61 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package grub provides the interface to the GRUB bootloader: config management, installation, etc.
+package grub
+
+import (
+	"os"
+
+	"github.com/siderolabs/go-blockdevice/blockdevice/probe"
+
+	"github.com/siderolabs/talos/internal/pkg/mount"
+	"github.com/siderolabs/talos/pkg/machinery/constants"
+)
+
+// Probe probes a block device for GRUB bootloader.
+func Probe() (*Config, error) {
+	// attempt to probe BOOT partition directly
+	dev, err := probe.GetDevWithPartitionName(constants.BootPartitionLabel)
+	if os.IsNotExist(err) {
+		// no BOOT partition, nothing to do
+		return nil, nil
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	defer dev.Close() //nolint:errcheck
+
+	mp, err := mount.SystemMountPointForLabel(dev.BlockDevice, constants.BootPartitionLabel)
+	if err != nil {
+		return nil, err
+	}
+
+	// if no BOOT partition nothing to do
+	if mp == nil {
+		return nil, nil
+	}
+
+	alreadyMounted, err := mp.IsMounted()
+	if err != nil {
+		return nil, err
+	}
+
+	if !alreadyMounted {
+		if err = mp.Mount(); err != nil {
+			return nil, err
+		}
+
+		defer mp.Unmount() //nolint:errcheck
```

Comment:
I'm a bit torn about defer blocks which ignore errors - we have a lot of them. We can do the named return error trick like:

```suggestion
		defer func() {
			unmountErr := mp.Unmount()
			if unmountErr != nil {
				err = unmountErr
			}
		}
```

Here and above, but I'm not sure if we can do anything meaningful on `Unmount` error. But at the same time, should we continue the execution as if nothing happened?

Reply:
this way we trade a (probably) unimportant error for a real failure of a function.. the better approach might be to make `.Unmount` idempotent and call it explicitly on "success" path with `return grubConf, mp.Unmount()`